services/media/service/src/ingest/handlers/check-finish-ingest-document-handler.ts

```diff
@@ -1,89 +1,79 @@
-import { LoginPgPool, transactionWithContext } from '@axinom/mosaic-db-common';
-import { Broker, MessageInfo } from '@axinom/mosaic-message-bus';
-import { sleep } from '@axinom/mosaic-service-common';
+import { Logger } from '@axinom/mosaic-service-common';
+import {
+  StoreOutboxMessage,
+  TypedTransactionalMessage,
+} from '@axinom/mosaic-transactional-inbox-outbox';
 import {
   CheckFinishIngestDocumentCommand,
   MediaServiceMessagingSettings,
 } from 'media-messages';
-import { SubscriptionConfig } from 'rascal';
+import { ClientBase } from 'pg';
 import { IngestItemStatusEnum, IngestStatusEnum } from 'zapatos/custom';
-import {
-  IsolationLevel,
-  param,
-  self as value,
-  sql,
-  SQL,
-  update,
-} from 'zapatos/db';
+import { param, self as value, sql, SQL, update } from 'zapatos/db';
 import { Config } from '../../common';
-import { MediaGuardedMessageHandler } from '../../messaging';
-
-interface StatusAggregation {
-  status: IngestItemStatusEnum;
-  count: number;
-}
+import { MediaGuardedTransactionalInboxMessageHandler } from '../../messaging';
+import { getFutureIsoDateInMilliseconds } from '../utils';
 
-export class CheckFinishIngestDocumentHandler extends MediaGuardedMessageHandler<CheckFinishIngestDocumentCommand> {
+export class CheckFinishIngestDocumentHandler extends MediaGuardedTransactionalInboxMessageHandler<
+  CheckFinishIngestDocumentCommand,
+  Config
```

Comment:
I think it should be possible to remove this `Config` generic parameter by changing the `MediaGuardedTransactionalInboxMessageHandler` in approximately the following way?
![image](https://github.com/Axinom/mosaic-media-template/assets/117624390/d0d20497-8d47-42b2-8f8a-c63f8c41ab2a)